pydantic/_internal/_generate_schema.py

```diff
@@ -482,6 +482,54 @@ def ser_ip(ip: Any, info: core_schema.SerializationInfo) -> str | IpType:
             },
         )
 
+    def _path_schema(self, tp: Any, path_type: Any) -> CoreSchema:
+        if tp is os.PathLike and path_type not in {str, bytes, Any}:
```

Comment:
We need to account for `typing.Any` and `typing_extensions.Any`, something like:

```suggestion
        if tp is os.PathLike and path_type not in {str, bytes} or not _typing_extra.is_any(tp):
```